tests/integration/test_random_creation.py

```diff
@@ -33,6 +35,366 @@ def test_randn():
+    assert rng_np_1.random() == rng_np_2.random()
+    assert rng_num_1.random() == rng_num_2.random()
+
+
+def test_default_rng_bitgenerator():
+    seed = 12345
+    rng_np_1 = np.random.default_rng(np.random.PCG64(seed))
+    rng_num_1 = num.random.default_rng(num.random.XORWOW(seed))
+    rng_np_2 = np.random.default_rng(np.random.PCG64(seed))
+    rng_num_2 = num.random.default_rng(num.random.XORWOW(seed))
+    assert rng_np_1.random() == rng_np_2.random()
+    assert rng_num_1.random() == rng_num_2.random()
+
+
+def test_default_rng_generator():
+    steps = 3
+    seed = 12345
+    seed_rng_np, seed_rng_num = gen_random_from_both("default_rng", seed)
+    np_rngs = [np.random.default_rng(seed_rng_np) for _ in range(steps)]
+    num_rngs = [num.random.default_rng(seed_rng_num) for _ in range(steps)]
+    np_seq_1 = [rng.random() for rng in np_rngs]
+    num_seq_1 = [rng.random() for rng in num_rngs]
+
+    rng_np, rng_num = gen_random_from_both("default_rng", seed)
+    np_seq_2 = [rng_np.random() for _ in range(steps)]
+    num_seq_2 = [rng_num.random() for _ in range(steps)]
+    assert np_seq_1 == np_seq_2
+    assert num_seq_1 == num_seq_2
+
+
+@pytest.mark.parametrize("shape", [(20, 2, 31), (10000,)], ids=str)
+def test_rand(shape):
+    arr_np = np.random.rand(*shape)
+    arr_num = np.random.rand(*shape)
+    assert arr_np.__class__.__name__ == arr_num.__class__.__name__
+    if hasattr(arr_np, "__array__"):
+        assert arr_np.shape == arr_num.shape
+        assert_distribution(
+            arr_num, np.mean(arr_np), np.std(arr_np), mean_tol=0.05
+        )
+
+
+LOW_HIGH = [
+    (5, 10000),
+    (-10000, 5),
+    pytest.param(
+        3000.45,
+        15000,
+        marks=pytest.mark.xfail(
+            reason="NumPy pass, cuNumeric hit struct.error"
+        ),
+        # When size is greater than 1024, legate core throws error
+        # struct.error: required argument is not an integer
+    ),
+    (10000, None),
+]
+SIZES = [
+    10000,
+    pytest.param(
+        None,
+        marks=pytest.mark.xfail(
+            reason="NumPy returns scalar, cuNumeric returns 1-dim array"
+            # with dtype:
+            # np.random.randint(5, 10000, None, np.int64).ndim  << 0
+            # num.random.randint(5, 10000, None, np.int64).ndim << 1
+            # without dtype:
+            # np.random.randint(5, 10000, None).__class__
+            #     <class 'int'>
+            # num.random.randint(5, 10000, None).__class__
+            #     <class 'cunumeric.array.ndarray'>
+        ),
+    ),
+    0,
+    (20, 50, 4),
+]
+
+
+@pytest.mark.parametrize("low, high", LOW_HIGH, ids=str)
+@pytest.mark.parametrize("size", SIZES, ids=str)
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        np.int64,
+        pytest.param(
+            np.uint16,
+            marks=pytest.mark.xfail(
+                reason="cuNumeric raises NotImplementedError"
+            ),
+        ),
+    ],
```

Comment:
would it make sense to include other additional int dtypes?

Reply:
Something, somewhere in the resulting codepaths will have to condition on dtype? Just based on experience, it's easy to imagine a weird bug that causes `cunumeric.random.randint` to explode of silently fail on just one dtype but not others.